mopidy_listenbrainz/frontend.py

```diff
@@ -132,16 +136,39 @@ def _collect_playlist_tracks(
         self, playlist_data: PlaylistData
     ) -> List[Track]:
         tracks: List[Track] = []
-        search_schemes = self.config["listenbrainz"].get(
+        search_schemes_mbid = self.config["listenbrainz"].get(
             "search_schemes", ["local:"]
         )
+        search_schemes_fallback = self.config["listenbrainz"].get(
+            "search_schemes_fallback", ["local:"]
+        )
 
         for track_mbid in playlist_data.track_mbids:
+            # try in the library first by MB id
             query = self.library.search(
-                {"musicbrainz_trackid": [track_mbid]}, uris=search_schemes
+                {"musicbrainz_trackid": [track_mbid]}, uris=search_schemes_mbid
             )
             results = query.get()
 
+            found_tracks = [t for r in results for t in r.tracks]
+            if len(found_tracks) == 0:
+                # retrieve track information from MB
+                mb_recording_query = musicbrainzngs.get_recording_by_id(track_mbid, includes=['artists'])
+                if mb_recording_query and mb_recording_query['recording']:
+                    mb_recording = mb_recording_query['recording']
+
+                    # try again with album artist name and track title
+                    artist_name = mb_recording['artist-credit-phrase']
+                    track_name = mb_recording['title']
+                    query = self.library.search({
+                            # very few backends support artist+track name queries,
+                            # so we use a keyword search, although it will be less precise
+                            'any': ['{} - {}'.format(artist_name, track_name)]
```

Comment:
Using "{artist} - {title}" - I don't know if this is the proper way.

Reply:
I need to set up a non-local backend before I do more testing, but:

From [docs](https://docs.mopidy.com/stable/api/core/#mopidy.core.LibraryController.search):

> ```py
# Returns results matching 'a' and 'b' and artist 'xyz' in any
# backend
search({'any': ['a', 'b'], 'artist': ['xyz']})
```

So maybe 
```suggestion
                            'any': [artist_name, track_name]
```

The docs make it sound like `any` must match everything in the list, but we would need to do more testing to confirm this.

Reply:
From brief testing it seems to work like an `AND` and not an `OR`, so what do you think about trying this suggestion? Make sure it works for you.

Reply:
I noticed that Mopidy (or mopidy-spotify probably, I didn't check) splits the query string anyway and creates many "any:word" tokens even if it's a single string, so I guess it doesn't really make much of a difference. But it does seem more correct so I'm going to follow your suggestion.